src/gapstate.c

```diff
@@ -5,30 +5,91 @@
  **
  ** This file declares all variables that are considered state for the
  ** interpreter
+ ** The background is that in HPC-GAP this state has to be thread local,
+ ** that is, an instance of the state has to exist for every thread, and
+ ** global variables are unsuitable for this purpose (even if protected
+ ** from concurrent access by locks).
```

Comment:
... and so this comment is IMHO confusing, too. Perhaps it should be moved to the header file instead?